### PR TITLE
Refactor and augment input region tests

### PR DESCRIPTION
**Pull request by wmww**
There's still a lot going on here, but hopefully this is more clear

### CODE REVIEW

**Comment by RAOF** [comment]
CI failure is genuine; FTBFS on 16.04

File: tests/surface_input_regions.cpp [hunk 1]
```diff
@@ -23,40 +23,43 @@
...
 
     std::string name;
+    std::pair<int, int> surface_size;
     std::vector<Element> elements;
 
     void apply_to_surface(wlcs::Client& client, wl_surface* surface) const
     {
+        if (elements.empty())
+            return;
         auto const wl_region = wl_compositor_create_region(client.compositor());
```
Comment (RAOF) [inline tests/surface_input_regions.cpp:53]: While we're here, does this want to be
`auto const wl_region = wlcs::wrap_wl_object(wl_compositor_create_region(client.compositor());`

Comment (wmww) [reply tests/surface_input_regions.cpp:53]: That could work, but this use case is simple enough to not really need it

**Comment by wmww** [comment]
Whoops, looks like this one slid through the cracks
bors r+

**Pull request merged**